scripts/babel/key-mirror-to-literal.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+module.exports = function keyMirrorToLiteral(babel) {
+  const t = babel.types;
+
+  function isKeyMirrorRequireCall(node) {
+    return (
+      node &&
+      node.type === 'CallExpression' &&
+      node.callee.type === 'Identifier' &&
+      node.callee.name === 'require' &&
+      node.arguments.length === 1 &&
+      node.arguments[0].type === 'StringLiteral' &&
+      (
+        node.arguments[0].value === 'fbjs/lib/keyMirror' ||
+        node.arguments[0].value === 'keyMirror'
+      )
+    );
+  }
+
+  return {
+    visitor: {
+      Program(path, file) {
+        const binding = path.scope.getBinding('keyMirror');
+        if (binding == null) {
+          return;
+        }
+        const initNode = binding.path.get('init').node;
+        if (!isKeyMirrorRequireCall(initNode)) {
+          return;
+        }
+        for (const reference of binding.referencePaths) {
+          if (!reference.parentPath.isCallExpression()) {
+            throw reference.buildCodeFrameError('Expected keyMirror function call.');
+          }
+          const argPath = reference.parentPath.get('arguments.0');
+          if (!argPath.isObjectExpression()) {
+            throw argPath.buildCodeFrameError('Expected object literal.');
+          }
+          for (const prop of argPath.get('properties')) {
+            const value = prop.get('value');
+            if (value.isNullLiteral()) {
+              const name = prop.get('key.name').node;
+              value.replaceWith(t.stringLiteral(name));
```

Comment:
Correct me if I’m wrong but I think this goes against the purpose of `keyMirror`.
It exists for Google Closure Compiler property mangling to work.
React is used by Om (and other libraries) from ClojureScript, which in turn relies on GCC.

AFAIK it will break Google Closure Compiler optimizations because

```js
keyMirror({ hello: true })
```

will get transpiled to

```
{ hello: 'hello' }
```

with this plugin (correct me if I’m wrong).

GCC will then mangle this to

```js
{ a: 'hello' }
```

which is broken.

The intention of `keyMirror` is that

```js
keyMirror({ hello: true })
```

gets mangled by GCC to

```js
```js
keyMirror({ a: true })
```

which will still work (because it produces `{ a: 'a' }`).